akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -385,27 +390,49 @@ private[http] object HttpServerBluePrint {
                 oneHundredContinueResponsePending = true
                 r.copy(createEntity = with100ContinueTrigger(r.createEntity))
               } else r
+              lastRequestOnConnectionSummary = s"${r.method.value} ${r.uri.path}"
               push(requestPrepOut, rs)
+
             case MessageEnd ⇒
               messageEndPending = false
               push(requestPrepOut, MessageEnd)
-            case MessageStartError(status, info) ⇒ finishWithIllegalRequestError(status, info)
+
+            case MessageStartError(status, info) ⇒
+              finishWithIllegalRequestError(status, info)
+
             case x: EntityStreamError if messageEndPending && openRequests.isEmpty ⇒
               // client terminated the connection after receiving an early response to 100-continue
               completeStage()
+
             case x ⇒
               push(requestPrepOut, x)
           }
-        override def onUpstreamFinish() =
+
+        override def onUpstreamFinish =
           if (openRequests.isEmpty) completeStage()
           else complete(requestPrepOut)
+
+        override def onUpstreamFailure(ex: Throwable): Unit = {
+          ex match {
+            case t: TimeoutException ⇒
+              // FIXME: this must be `akka.stream.impl.io.TcpIdleTimeoutException` instead! Then improve below exception:
+              log.debug(
+                "Timeout exception occured while replying to request: [{}], shutting down connection. Cause: {}",
```

Comment:
The timeout originates in Akka infrastructure, so there's no sensible way to attach "akka.http setting..." information in here, thus we need to know by exception type that it was the idle timeout, and not a random timeout, so we can log an useful message.